src/xapi.c

```diff
@@ -1148,21 +1147,35 @@ int xapi_variables_request(variable_t *vars, size_t n)
         return 0;
     }
 
+    plaintext = (uint8_t*)malloc(MSG_SIZE * 2);
+    if (!plaintext) {
+        ERROR("failed to allocate memory\n");
+        return -ENOMEM;
+    }
+
+    b64  = (char*)(plaintext + MSG_SIZE);
+    memset(plaintext, 0, MSG_SIZE * 2);
+
     ret = xapi_get_nvram(session_id, b64, MSG_SIZE);
 
     if (ret < 0) {
-        return 0;
+        ret = 0;
+        goto out;
     }
 
     session_logout(session_id);
 
     ret = base64_to_bytes(plaintext, MSG_SIZE, b64, strlen(b64));
 
     if (ret < 0) {
-        return 0;
+        goto out;
```

Comment:
`ret = 0`?

Reply:
Yep . This is wierd, but originally allmost all errorcases return 0. ??? I leaved this case, because may be it's tolerated to not get nvram for guest (can't contact XAPI backend). Normally obviously all errors should return errors. But in the original Bobbye work it is not true